spec/wegift/models/remote_code_spec.rb

```diff
@@ -8,6 +8,43 @@
     let(:remote_code) { client.remote_code(url) }
     let(:client) { set_wegift_client }
 
+    context 'when URL is invalid' do
+      let(:url) { 'https://thisnotworking.yo' }
+      
+      it 'raises error' do
+        VCR.use_cassette('get_remote_code_invalid_url_does_not_exist') do
+          expect { remote_code }.to raise_error(Faraday::ClientError)
+        end
+      end
+    end
+
+    context 'when URL is valid but not known' do
+      let(:url) { 'https://example.com' }
+      
+      it 'is not successful' do
+        VCR.use_cassette('get_remote_code_invalid_wrong_url_exists') do
+          expect(remote_code.is_successful?).to be false
+          # Yeah, right this URL can be ANYTHING, so client has to be careful!
+          # Remote code is not successful, because the parser didn't find the SUCCESS response!
+          # But the actual request itself is of course a HTTP 200
+          expect(remote_code.error_code).to eq 200
+          expect(remote_code.error_string).to eq "OK"
+        end
+      end
+    end
+
+    context 'when URL is valid but not known to wegift' do
+      let(:url) { 'https://playground.wegift.io/public/gifts/instant/c02bd09f-0000-0000-0000-38f4143a01d1' }
+      
+      it 'is not successful' do
+        VCR.use_cassette('get_remote_code_invalid_unknown_wegift_url') do
+          expect(remote_code.is_successful?).to be false
+          expect(remote_code.error_code).to eq 404
+          expect(remote_code.error_string).to eq "Not Found"
```

Comment:
I think this is a problem, it doesn't follow the same error logic as the other endpoints, as in:

* return 200 HTTP code
* set the error in the body with the `error_code`, `error_details` and `error_string`

So any client relying on the error message won't get anything more than the HTTP status (and the message associated)

Reply:
We could ask them, if we can assume any URL is on their side and under their control.

Reply:
it seems to be consistent "as possible" lol

Reply:
it  does return a body with the actual error code defined in their documentation but the status code is not 200 unfortunately :/

Reply:
Yeah, because the URL could be anything, so it's correct from them to behave like that and not like the Wegift API...

Reply:
I think there's still a difference (even in their implementation) between a wrong URL and a wrong e-code UUID, as the `format` query parameter seems to be only applied to the `https://playground.wegift.io/public/gifts/instant` path prefix. The former returns a `text/html` with a 404 HTTP status while the latter returns an `application/json` with the correct body and the appropriate error details in the body but still, a 404 HTTP status code.

Reply:
So what are you suggesting here?

Reply:
It will sound horrible but to try and have the same behaviour, we would need to implement a different error handling logic for this specific method/endpoint, taking into account the existence or not of these json attributes regardless of the status code.
It would seem that as long as the content type is `application/json`, the body of the response will contain json. If it returns a 404 because the address is incorrect, then wegift returns a `text/html` content type.

Reply:
OK. Since this change is actually just adding some tests for the current behavior, I would say, we treat it as a separate issue.